### PR TITLE
Fixing an issue when using input-bit with qoqo-for-braket

### PR DESCRIPTION
**Pull request by nfwvogt**
This pull request:

* Fixes an issue where the backend failed when InputBit operations are present in a Circuit to set the bits in a register for indexes that are NOT read from qubits. The backend now filters out InputBit operations and sets the bits in post-processing
* Fixes deviation from other backends that only allowed using `.run_program(backend, [param_list1, param_list2,...])` not allowing to run with a single list of parameters. Now `.run_program(backend, param_list)` is also supported.

### CODE REVIEW

File: qoqo_for_braket/qoqo_for_braket/post_processing.py [hunk 1]
```diff
@@ -13,7 +13,9 @@
 from typing import Dict, List, Tuple, Any
 
 
-def _post_process_circuit_result(results: Any, metadata: Dict[Any, Any]) -> Tuple[
+def _post_process_circuit_result(
+    results: Any, metadata: Dict[Any, Any], input_bit_circuit
+) -> Tuple[
```
Comment (mlodi-hqs) [inline qoqo_for_braket/qoqo_for_braket/post_processing.py:18]: Same thing here

File: qoqo_for_braket/qoqo_for_braket/post_processing.py [hunk 2]
```diff
@@ -37,6 +39,30 @@ def _post_process_circuit_result(results: Any, metadata: Dict[Any, Any]) -> Tupl
...
+        )
+    ]
+
+    # Appending extension bits to each measured result
+    for key, value in bit_results.items():
+        bit_results_final[key] = []
+        for val in value:
+            bit_results_final[key].append(val.extend(extension_bits))
 
     return (bit_results, float_results, complex_results)
```
Comment (mlodi-hqs) [inline qoqo_for_braket/qoqo_for_braket/post_processing.py:68]: Shouldn't this then return `bit_results_final`?

**Comment by mlodi-hqs** [comment]
I'll fix the non-working tests.

**Review by mlodi-hqs — CHANGES_REQUESTED** [review]

File: qoqo_for_braket/qoqo_for_braket/post_processing.py (hunk 2 above)
Comment (nfwvogt) [reply qoqo_for_braket/qoqo_for_braket/post_processing.py:68]: You are absolutely right. Could you also apply that fix?

**Review by mlodi-hqs — APPROVED** [review]

**Pull request merged**